airflow/utils/log/file_task_handler.py

```diff
@@ -357,7 +367,9 @@ def _read(
             worker_log_full_path = Path(self.local_base, worker_log_rel_path)
             local_messages, local_logs = self._read_from_local(worker_log_full_path)
             messages_list.extend(local_messages)
-        if is_running and not executor_messages:
+        if ti.task.inherits_from_empty_operator is True and self._test_mode is False:
```

Comment:
Why does it matter if we are in test mode or not?

Reply:
So this change actually causes several tests to fail because many tests check to see that the following call (`served_messages, served_logs = self._read_from_logs_server(ti, worker_log_rel_path`)) is made. Let me take another look at this and see if we can do the override for EmptyOperator while still keeping these tests intact.

Reply:
Okay, modified my approach here to catch the exception instead which is much less disruptive to tests. Retested manually as well and all continues to look good - 

<img width="1470" alt="Screenshot 2024-01-11 at 9 44 50 AM" src="https://github.com/apache/airflow/assets/9200263/81b36557-4841-45ae-b8c0-f6d7c3563d8d">